### DISCUSSION TITLE
[Question]: Frigate+ object training

### DISCUSSION THREAD

**Discussion by simenopsahl** [blakeblackshear/frigate#10945] [Ask A Question] [answered]
### What is your question:

I have a question regarding how images are used in the model training. I am currently tracking (cats, dogs, people, packages) in this camera view. I am not interested in tracking cars, but they sometimes appear in the view.

When I am submitting images, is the model trained on the whole image with special attention to the marked areas? Or is every individual marked object cropped out and submitted to the model independently?

Currently I am tagging cars and license plates also, but if I were to omit those objects from marking would the public model suffer as a result? When I am going through and tagging (sometimes hundreds) of submissions, could I save myself the trouble of tagging cars and license plates without impacting other users?
![2024-04-11 203834](https://github.com/blakeblackshear/frigate/assets/2676016/c3aeb427-0e9e-40b5-9ed3-7801cb06eef7)

**Comment by NickM-27** [comment, marked as answer]
> When I am submitting images, is the model trained on the whole image with special attention to the marked areas? Or is every individual marked object cropped out and submitted to the model independently?

neither of these. Parts of the image that have objects marked are cropped but it is not necessarily each object. For example those people are close together so it is possible that a single crop would be done including both of those objects. The goal of the model training is to match the way that frigate runs object detection.

You can see this by using the debug live view in the frigate UI, `regions` are the areas sent to detection

> When I am going through and tagging (sometimes hundreds) of submissions, could I save myself the trouble of tagging cars and license plates without impacting other users?

user images are not automatically included in the base model, they are reviewed and corrected by Blake. So by omitting cars you are only potentially affecting the results of your own model.

**Reply by simenopsahl** [reply]
Thank you, then I will save so much time 😅
I currently have about ~9000 submitted images fully tagged from that camera, in rain, snow, night etc. But I will stop tagging cars from now on.
